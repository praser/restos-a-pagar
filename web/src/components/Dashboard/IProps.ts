interface IProps {
  anoOrcamentario: number;
  dataBloqueio: string;
}

export default IProps;
