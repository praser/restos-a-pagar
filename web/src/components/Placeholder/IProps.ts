interface IProps {
  src: string;
}

export default IProps;
