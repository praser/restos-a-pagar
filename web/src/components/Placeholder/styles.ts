import styled from 'styled-components';

const Container = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;

const Image = styled.img`
  height: 60%;
`;

export { Container, Image };
