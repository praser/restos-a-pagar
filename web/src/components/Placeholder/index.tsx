import React, { FunctionComponent } from 'react';

import IProps from './IProps';
import { Container, Image } from './styles';

const Placeholder: FunctionComponent<IProps> = ({ src, children }) => {
  return (
    <>
      {children}
      <Container>
        <Image src={src} />
      </Container>
    </>
  );
};

export default Placeholder;
