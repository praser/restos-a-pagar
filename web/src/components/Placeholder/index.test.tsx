import React from 'react';

import renderer from 'react-test-renderer';

import IProps from './IProps';
import Placeholder from './index';

import image from 'assets/undraw_well_done_i2wr.svg';

describe('Placeholder Component', () => {
  let props: IProps;
  beforeAll(() => {
    props = {
      src: image,
    };
  });

  it('is expected to render the component correctly', () => {
    const component = renderer.create(<Placeholder {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
