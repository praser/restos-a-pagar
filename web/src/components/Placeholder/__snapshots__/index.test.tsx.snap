// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder Component is expected to render the component correctly 1`] = `
<div
  className="sc-bdfBwQ jxmVCH"
>
  <img
    className="sc-gsTCUz hFcGpJ"
    src="undraw_well_done_i2wr.svg"
  />
</div>
`;
